File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.cs
public class Solution {
    public int LongestConsecutive(int[] nums) {
        if(nums.Length == 0) return 0;
        Array.Sort(nums);
        
        int longestStreak = 1;
        int currentStreak = 1;

        for(int i = 1; i < nums.Length; i++){
            if(nums[i] != nums[i-1]){
                if(nums[i] == nums[i-1] + 1){
                    currentStreak++;
                }else{
                    longestStreak = Math.Max(longestStreak, currentStreak);
                    currentStreak = 1;
                }
            }
        }

        return Math.Max(longestStreak, currentStreak);
    }
}

File: Data Structures & Algorithms/house-robber-ii/submission-2.java
class Solution {
    public int rob(int[] nums) {
        
        if(nums.length == 1){
            return nums[0];
        }else if(nums.length == 2){
            return Math.max(nums[0], nums[1]);
        }

        int maxRouteA = checkMaxPrice(nums, 0, nums.length -1);
        int maxRouteB = checkMaxPrice(nums, 1, nums.length);
        return Math.max(maxRouteA, maxRouteB);
    }

    public int checkMaxPrice(int[] nums, int startIndex, int endIndex){
        int prevPrevHouse = nums[startIndex];
        int prevHouse = Math.max(prevPrevHouse, nums[startIndex + 1]);
        int maxProfit = prevHouse;

        for (int i = startIndex + 2; i < endIndex; i++) {
            maxProfit = Math.max(prevHouse, prevPrevHouse + nums[i]);
            prevPrevHouse = prevHouse;
            prevHouse = maxProfit;
        }

        return maxProfit;
    }
}
